template/express-rest-api/Dockerfile

```diff
@@ -1,15 +1,24 @@
-ARG BASE_IMAGE
-ARG BASE_TAG
+# syntax=docker/dockerfile:1.2
+
+FROM node:16-alpine AS dev-deps
+
+WORKDIR /workdir
+
+COPY package.json yarn.lock ./
+
+RUN \
+  --mount=type=secret,id=npm,dst=/workdir/.npmrc \
+  yarn install --frozen-lockfile --ignore-optional --non-interactive
```

Comment:
Oh, I just remembered why we were forced to go with two Dockerfiles: Gantry doesn't currently support build secrets, so it _can't_ run this `dev-deps` stage.

We could remove the `syntax` and `mount` instructions given the templates only use public packages, but it will mean that devs will invariably need to read our doco and split their Dockerfile once they need to consume private packages.

Alternatively, we could look into landing build secret support in Gantry first so we can land this PR as-is. I had a poke around today and it seems like we can achieve this by flipping `ImageBuild` into BuildKit mode, but it's definitely a non-trivial change.